src/codemagic/tools/app_store_connect/abstract_base_action.py

```diff
@@ -55,6 +77,21 @@ class AbstractBaseAction(ResourceManagerMixin, PathFinderMixin, metaclass=ABCMet
     _issuer_id: Optional[IssuerId]
     _private_key: Optional[str]
 
+    @staticmethod
+    def _get_certificate_key(
+        certificate_key: Optional[Union[PrivateKey, Types.CertificateKeyArgument]] = None,
+        certificate_key_password: Optional[Types.CertificateKeyPasswordArgument] = None,
+    ) -> Optional[PrivateKey]:
+        if isinstance(certificate_key, PrivateKey):
+            return certificate_key
+        password = certificate_key_password.value if certificate_key_password else None
+        if certificate_key is not None:
+            try:
+                return PrivateKey.from_buffer(certificate_key.value, password)
+            except ValueError:
+                CertificateArgument.PRIVATE_KEY.raise_argument_error("Not a valid certificate private key")
+        return None
```

Comment:
It used to be defined in https://github.com/codemagic-ci-cd/cli-tools/blob/3670fa624997f17cb437bc88198f7bdee3594f35/src/codemagic/tools/app_store_connect.py#L82